include/taskolib/GitRepository.h

```diff
@@ -0,0 +1,155 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+
+#include <git2.h>
+#include <string>
+#include <filesystem>
+#include "taskolib/GitObjectWrapper.h"
+#include <vector>
+
+namespace task {
+
+
+/**
+ * A class to wrap used methods from C-Library libgit2
+ * 
+ * This class is necessary to handle C-pointers in a way that
+ * C++ programmers using this class do not have to think about them.
+ * 
+ * This class does not reflect the full libgit2 library.
+ * git functions which are not implemented in this class will be
+ * implemented in case of necessity
+*/
+class GitRepository
+{
+public:
+
+    /**
+     * Constructor which specifies the root dir of the git repository
+     * \param file_path: path to "sequences"-folder (or customized directory)
+     *
+     */
+    explicit GitRepository(std::filesystem::path file_path);
+
+    /// returns member variable, which is the root dir of the git repository
+    std::filesystem::path get_path() const;
+
+    /// returns a C-type pointer to the current loaded repository
+    git_repository* get_repo();
+
+    /// stage all new and changed files in the repository environment
+    void libgit_add();
+
+    /**
+     * 
+    */
+   std::vector <int> libgit_add_files(std::vector<std::filesystem::path> filepaths);
+
+    /**
+     * return the commit message of ther HEAD commit
+     * \note this function is solely for Unittest purposes
+     * \return message of last commit (=HEAD)
+    */
+    std::string get_last_commit_message();
+
+    /**
+     * Commit staged changes to the master branch of the git repository
+     * \param commit_message: customized message for the commit
+    */
+    void libgit_commit(const std::string& commit_message);
+
+    /**
+     * Deletes seq_repository and all files within
+     * \param seq_directory: directory holding sequence
+     * 
+     * \attention seq_directory is a relative path with repo_path_ as root
+     *            if the sequence to be deleted is in "sequences/"
+     *            then set seq_directory = SEQUENCE_NAME
+     * 
+    */
+    void libgit_remove_sequence(std::filesystem::path seq_directory);
+
+    /// returns current git
+    std::vector<std::array<std::string, 3>> libgit_status();
+
+    /// Destructor
+    ~GitRepository();
+
+private:
+    /// Pointer which holds all infos of the active repository
+    LgObject<git_repository*> repo_{ nullptr };
+
+    /// path to the repository (for taskomat .../sequences/)
+    std::filesystem::path repo_path_;
+
+    /// signature used in commits
+    LgObject<git_signature*> my_signature_{ nullptr };
```

Comment:
This is what I meant with `T` above: It might be nicer if this could be written
```suggestion
    GitPointer<git_signature> my_signature_{ nullptr };
```

Reply:
It is not intended to have those dataypes "bare" in libgit2. Therefore I recommend to also construct the wrappers with the pointer type